core/src/kura.rs

```diff
@@ -694,6 +844,22 @@ pub enum Error {
     Locked(PathBuf),
     /// Conversion of wide integer into narrow integer failed. This error cannot be caught at compile time at present
     IntConversion(#[from] std::num::TryFromIntError),
+    /// Blocks count differs hashes file and index file
+    HashesFileHeightMismatch,
+}
+
+trait AddErrContextExt<T> {
```

Comment:
This feels like re-implementing [`snafu`](https://docs.rs/snafu/latest/snafu/).. Probably fine for one file, be we should consider using `snafu`..

Reply:
Good point, let's consider this refactoring if such pattern will appear in other places too

Reply:
What I don't understand about snafu is that it adds `Snafu` suffix to types. Why? It looks ugly to me

Reply:
`Snafu` suffix is for context helpers - all values for your error variant, excluding the error itself. Instead of writing `res.map_err(|e| Error::IoError { error: e, path })` you would write something akin to `res.context(|| IoSnafu { path })`.

It also obviously can derive `From` if your error variant does not need any context. Or you can use the `Snafu` struct to disambiguate between two variants